lyluatex.lua

```diff
@@ -770,62 +839,22 @@ function Score:run_lilypond()
-    if self.printfilename and self.input_file then
-        if self.insert == 'fullpage' then
-            warn('`printfilename` ignored with `insert=fullpage`')
-        else
-            local filename = self.input_file:gsub("(.*/)(.*)", "\\lyFilename{%2}\\par")
-            tex.sprint(filename)
-        end
-    end
-    if self.verbatim then
-        ly.verbprint(self.orig_ly_code:explode('\n'))
-        if self.intertext then
-            tex.print('\\lyIntertext{'..self.intertext..'}\\par')
-        end
-    end
+function Score:write_latex(do_compile)
+    latex.filename(self.printfilename, self.insert, self.input_file)
+    latex.verbatim(self.verbatim, self.orig_ly_code, self.intertext, self.addversion)
     if do_compile then
-        if not self:check_failed_compilation() then return end
+        if not self:is_compiled_without_error() then return end
     end
     --[[ Now we know there is a proper score --]]
-    if self.fullpagestyle == '' then
-        if self['print-page-number'] then
-            set_fullpagestyle('empty')
-        else set_fullpagestyle(nil)
-        end
-    else set_fullpagestyle(self.fullpagestyle)
-    end
-    local label = ''
-    if self.label then label = '\\label{'..self.labelprefix..self.label..'}' end
-    local systems_file = io.open(self.output..'-systems.tex', 'r')
-    if not systems_file then
-        --[[ Fullpage score, use \includepdf ]]
-        tex.sprint(label..'\\includepdf[pages=-]{'..self.output..'}')
-    else
-        --[[ Fragment, use -systems.tex file]]
-        local content = systems_file:read("*all")
-        local texoutput
-        systems_file:close()
-        if do_compile then
-            --[[ new compilation, calculate protrusion
-                 and update -systems.tex file]]
-            local protrusion = self:protrusion()
-            texoutput = content:gsub([[\includegraphics{]],
-                [[\noindent]]..' '..protrusion..[[\includegraphics{]]..dirname(self.output))
-            local f = io.open(self.output..'-systems.tex', 'w')
-            f:write(texoutput)
-            f:close()
-            self:delete_intermediate_files()
-        else
-            -- simply reuse existing -systems.tex file
-            texoutput = content
-        end
-        texoutput = label..
-            [[\ifx\preLilyPondExample\undefined\else\expandafter\preLilyPondExample\fi]]..
-            texoutput..
-            [[\ifx\postLilyPondExample\undefined\else\expandafter\postLilyPondExample\fi]]
-        tex.sprint(texoutput:explode('\n'))
+    latex.fullpagestyle(self.fullpagestyle, self['print-page-number'])
+    latex.label(self.label, self.labelprefix)
+    local systems_file = self.output..'-systems'
+    if not lfs.isfile(systems_file..'.tex') then  -- fullpage score
+        latex.includepdf(self.output)
+    else  -- fragment
+        latex.includesystems(
+            systems_file, self:_protrusion(), convert_unit(self.indent), do_compile
```

Comment:
This will fail in line 821 when `do_compile == false` (i.e. when a document is compiled the second time) because the `.eps` file has been removed after the first run.

I have to leave the computer now, therefore I can't  look for a fix.

Reply:
It didn't fail, because `_protrusion()` returned `nil` and this result wasn't used in `latex.includesystems()`. But I rewrote it in a more explicit way, and it gave me the occasion to enhance `latex.includesystems()`.

Reply:
It *did* (does) fail before `_protrusion()` gets a chance to return because there is no `.eps` file to open, therefore `f` is nil and `f:read()` fails.

Reply:
Minimal document:

```TeX
\documentclass{scrartcl}
\usepackage{lyluatex}
\begin{document}

\lilypond{ c' }

\end{document}
```

Failure upon *second* compilation:

```
lyluatex/lyluatex.
lua:821: attempt to index local 'f' (a nil value)
stack traceback:
	...e/uliska/texmf/tex/latex/latex-git/lyluatex/lyluatex.lua:821: in function '
_protrusion'
	...e/uliska/texmf/tex/latex/latex-git/lyluatex/lyluatex.lua:857: in function '
write_latex'
	...e/uliska/texmf/tex/latex/latex-git/lyluatex/lyluatex.lua:809: in function '
process'
	[\directlua]:1: in main chunk.
\ly@compilescore ...directlua {ly.score:process()}
                                                  
l.5 \lilypond{ c' }
                 
?
```

Reply:
That's weird, I didn't have that error. Never mind, it's fixed now.

Reply:
Thanks, works now.

I must admit I'm not 100% I've assessed and tested everything, but I think this PR is good to be merged now.